shiny/ui/_utils.py

```diff
@@ -92,3 +94,25 @@ def css_no_sub(**kwargs: str | float | None) -> Optional[str]:
         v = " ".join(v) if isinstance(v, list) else str(v)
         res += k + ":" + v + ";"
     return None if res == "" else res
+
+
+def extract_js_keys(options: Dict[str, Any], parent_key: str = "") -> List[str]:
+    """
+    This function extracts JavaScript (JS) and HTML objects from the provided dictionary.
+    This is used to identify which options should be evaluated by the client.
+
+    Parameters:
+    options (Dict[str, Any]): A dictionary containing various options. The options can be of any type,
+    but the function specifically looks for HTML, JS, and dictionary types.
+
+    Returns:
+    A list of keys which identify the JS or HTML options
+    """
+    js_html_keys = []
+    for key, value in options.items():
+        full_key = f"{parent_key}.{key}" if parent_key else key
+        if isinstance(value, (HTML, JS)):
```

Comment:
```suggestion
        if isinstance(value, JS):
```

Reply:
I used a new class `JSEval` for this instead.